fru.c

```diff
@@ -497,7 +517,30 @@ fru_info_area_t *fru_create_info_area(fru_area_type_t atype,    ///< [in] Area t
 	     field && !field->rec && field_count < nstrings;
 	     field = field->next, field_count++)
 	{
-		field->rec = fru_encode_data(LEN_AUTO, strings[field_count]);
+		int len = LEN_AUTO;
+		switch (strings[field_count].type) {
+			case FIELD_TYPE_BINARY: {
+				DEBUG("binary format not yet implemented\n");
+				len = LEN_AUTO;
```

Comment:
I believe we must bail out on that, setting `errno` to `ENOTSUP` or `EINVAL`, and going to `err`.

Reply:
Changed code accordingly.